apps/dev-portal-web/domains/docs/components/widgets/AmountDistributionCalculator/DistributionItem.tsx

```diff
@@ -0,0 +1,110 @@
+    const FeePayerTitle = intl.formatMessage({ id: 'docs.widgets.amountDistributionCalculator.isFeePayer' })
+    const VORTitle = intl.formatMessage({ id: 'docs.widgets.amountDistributionCalculator.vor' })
+    const OverpaymentTitle = intl.formatMessage({ id: 'docs.widgets.amountDistributionCalculator.overpaymentPart' })
+
+    const [tin, setTin] = useState<string>(item.recipient.tin)
+    const [bic] = useState<string>(item.recipient.bic)
+    const [bankAccount] = useState<string>(item.recipient.bankAccount)
+    const [amount, setAmount] = useState<string>(item.amount)
+    const [isFeePayer, setIsFeePayer] = useState<boolean | undefined>(item.isFeePayer)
+    const [order, setOrder] = useState<number | undefined>(item.order || 0)
+    const [vor, setVor] = useState<boolean | undefined>(item.vor)
+    const [overpaymentPart, setOverpaymentPart] = useState<number | undefined>(item.overpaymentPart)
+
+    useEffect(() => {
+        onUpdate(index, {
+            recipient: { tin, bic, bankAccount },
+            amount, isFeePayer, order, vor, overpaymentPart,
+        })
+    }, [amount, bankAccount, bic, index, isFeePayer, onUpdate, order, overpaymentPart, tin, vor])
+
+    return (
+        <Row align='middle' justify='space-between' gutter={GUTTER_COMPACT_ROW}>
+            <Col span={24}>
+                <Input
+                    value={tin}
+                    onChange={(e) => {
+                        setTin(e.target.value)
+                    }}
+                    prefix={`${TinTitle}:`}
+                />
+            </Col>
+            <Col span={24}>
+                <Input
+                    value={amount}
+                    onChange={(e) => {
+                        setAmount(e.target.value)
+                    }}
+                    prefix={`${AmountTitle}:`}
+                />
+            </Col>
+            <Col span={24}>
+                <Input
+                    value={order}
+                    onChange={(e) => {
+                        try {
+                            setOrder(Number(e.target.value))
+                        } catch (err) {
+                            setOrder(0)
+                        }
+                    }}
+                    prefix={`${OrderTitle}:`}
+                />
+            </Col>
+            <Col span={24}>
+                <Row gutter={0} align='middle' justify='space-between' className={styles.recipientCheckboxRow}>
+                    <Col><Typography.Text>{FeePayerTitle}:</Typography.Text></Col>
+                    <Col>
+                        <Switch size='large' checked={isFeePayer} onChange={(checked) => setIsFeePayer(checked)}/>
+                    </Col>
+                </Row>
+            </Col>
+            <Col span={24}>
+                <Row gutter={0} align='middle' justify='space-between' className={styles.recipientCheckboxRow}>
+                    <Col><Typography.Text>{VORTitle}:</Typography.Text></Col>
```

Comment:
It's a bad practice to:
a) Have col without explicit span
b) Have to many inline components

Why do you use row with zero-gutter and col with no span?

Col and Row combo used to define basic grid system, so `space-between` and `middle` seems a bit strange to me. What are trying to achieve? If its some flexbox layout why dont just use regular divs with css-modules instead?

Reply:
Refactored